[OPIK-337] anthropic integration

alexkuzmik opened the pull request:
## Details
This PR adds an integration with Anthropic similar to the ones we have for OpenAI and Bedrock.
Only non-streaming mode is supported at the moment.

Synchronous API
```python
import asyncio

from anthropic import Anthropic, AsyncAnthropic
from opik.integrations.anthropic import track_anthropic

client = Anthropic()
client = track_anthropic(client)

client.messages.create(
    max_tokens=1024,
    messages=[
        {
            "role": "user",
            "content": "Tell a fact",
        }
    ],
    model="claude-3-opus-20240229",
)


async_client = AsyncAnthropic()
async_client = track_anthropic(async_client)

asyncio.run(
    async_client.messages.create(
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": "Tell a joke",
            }
        ],
        model="claude-3-opus-20240229",
    )
)
```


## Issues

Resolves #

## Testing

## Documentation

Lothiraldan commented on sdks/python/src/opik/integrations/anthropic/opik_tracker.py:
```diff
@@ -0,0 +1,106 @@
+        )
+    except Exception:
+        LOGGER.debug(
+            "Failed to patch messages.batch method. It is likely because it was not implemented in the provided anthropic client",
```
```suggestion
            "Failed to patch messages.batches method. It is likely because it was not implemented in the provided anthropic client",
```

Lothiraldan commented on sdks/python/src/opik/integrations/anthropic/opik_tracker.py:
```diff
@@ -0,0 +1,106 @@
+        LOGGER,
+    )
+
+    anthropic_client.messages.create = create_decorator(
+        anthropic_client.messages.create
+    )
+    anthropic_client.messages.stream = stream_decorator(
+        anthropic_client.messages.stream
+    )
```
Same here about exception handling, that could fails if the anthropic SDK change its public API (unlikely)

alexkuzmik replied:
Agreed on relying on public API.

Lothiraldan reviewed (commented):
Looks good, some additional feedback:

* I'm getting an error when calling `messages.create(stream=True)`:
```
stream_manager = client.messages.create(
    model="claude-3-5-sonnet-20241022",
    max_tokens=1024,
    messages=[
        {"role": "user", "content": "Hello, Claude, tell me a very good joke"}
    ],
    stream=True,
)

with stream_manager as stream:
    for event in stream:
        print(event)
```

Do you know if that's the old API for `client.messages.stream`?

* Could you remind me if we reached a decision regarding the legacy API for completions? I am not getting any trace logged when calling the following code neither a warning:

```
print(client.completions.create(
    model="claude-2.1",
    max_tokens_to_sample=1024,
    prompt=f"{HUMAN_PROMPT} Hello, Claude{AI_PROMPT}",
))
```

alexkuzmik commented:
@Lothiraldan
Added support for `client.messages.create(stream=True).
Added a warning for `client.completions.create` calls.
Updated the PR description, now it has all supported usage examples.

Lothiraldan reviewed (approved):
LGTM
